Paintroid/src/org/catrobat/paintroid/tools/implementation/BaseToolWithRectangleShape.java

```diff
@@ -389,12 +393,14 @@ private void drawRotationArrows(Canvas canvas) {
 	private void drawBitmap(Canvas canvas) {
 
 		Paint bitmapPaint = new Paint(Paint.DITHER_FLAG);
+        canvas.save();
+
 		canvas.clipRect(new RectF(-mBoxWidth / 2, -mBoxHeight / 2,
-				mBoxWidth / 2, mBoxHeight / 2), Op.UNION);
-		canvas.drawBitmap(mDrawingBitmap, null, new RectF(-mBoxWidth / 2,
-				-mBoxHeight / 2, mBoxWidth / 2, mBoxHeight / 2), bitmapPaint);
+                mBoxWidth / 2, mBoxHeight / 2), Op.UNION);
+		canvas.drawBitmap(mDrawingBitmap, null, new RectF(-mBoxWidth / 2, -mBoxHeight / 2,
+                mBoxWidth / 2, mBoxHeight / 2), bitmapPaint);
 
-	}
+    }
 

```

Comment:
take a look at your code formating. The changes above are only formating changes.